packages/viewer/src/store/modules/position/actions/zoomToExtent.ts

```diff
@@ -0,0 +1,82 @@
+import type { FlatExtent, NormalizedExtent, SingleCoordinate } from '@swissgeo/coordinates'
+import type { Position } from 'geojson'
+
+import { CoordinateSystem, extentUtils, WGS84 } from '@swissgeo/coordinates'
+import { center, points } from '@turf/turf'
+import proj4 from 'proj4'
+
+import type { PositionStore } from '@/store/modules/position/types/position'
+import type { ActionDispatcher } from '@/store/types'
+
+import useUIStore from '@/store/modules/ui.store'
+
+export default function zoomToExtent(
+    this: PositionStore,
+    payload: {
+        extent: FlatExtent | NormalizedExtent
+        extentProjection?: CoordinateSystem
+        maxZoom?: number
+    },
```

Comment:
can you get rid of the payload (flatten in)?
you might want to push the optional into a new interface that may or may not be in the args (function overload)